float32[] hull_x
float32[] hull_y
float32[] centroid_x
float32[] centroid_y
